tests/Propulsion.MessageDb.Integration/Tests.fs

```diff
@@ -78,3 +88,47 @@ let ``It processes events for a category`` () = task {
     // they were handled in order within streams
     let ordering = handled |> Seq.groupBy fst |> Seq.map (snd >> Seq.map snd >> Seq.toArray) |> Seq.toArray
     test <@ ordering |> Array.forall ((=) [| 0L..19L |]) @> }
+
+type LogSink() =
+    let mutable items = 0
+    let mutable calledTimes = 0
+    member _.Events = items
+    member _.CallCount = calledTimes
+    interface ILogEventSink with
+        member _.Emit(logEvent) =
+            match logEvent with
+            | Propulsion.Feed.Core.Log.MetricEvent (Propulsion.Feed.Core.Log.Metric.Read r) ->
+                calledTimes <- calledTimes + 1
+                items <- items + r.items
+            | _ -> ()
+
+[<Fact>]
+let ``It doesn't read the tail event again`` () = task {
+    let logSink = LogSink()
+    let log = LoggerConfiguration().WriteTo.Sink(logSink).CreateLogger()
+    let consumerGroup = $"{Guid.NewGuid():N}"
+    let category = $"{Guid.NewGuid():N}"
+    let connString = "Host=localhost; Database=message_store; Port=5433; Username=message_store; Password=;"
+    use conn = new NpgsqlConnection(connString)
+    do! conn.OpenAsync()
+    do! writeMessagesToStream conn $"{category}-1"
+    do! conn.CloseAsync()
+    let checkpoints = ReaderCheckpoint.CheckpointStore("Host=localhost; Database=message_store; Port=5433; Username=postgres; Password=postgres", "public", $"TestGroup{consumerGroup}", TimeSpan.FromSeconds 10)
+    do! checkpoints.CreateSchemaIfNotExists()
+    let stats = stats log
+
+    let handle _ _ _ = task {
+        return struct (Propulsion.Streams.SpanResult.AllProcessed, ()) }
+    use sink = Propulsion.Streams.Default.Config.Start(log, 10, 1, handle, stats, TimeSpan.FromMinutes 1)
+    let source = MessageDbSource(
+        log, TimeSpan.FromMilliseconds 10,
+        connString, 1000, TimeSpan.FromMilliseconds 10,
+        checkpoints, sink, [| category |])
+    use src = source.Start()
+
+    // Fetch 3 page
+    while logSink.CallCount < 3 do ()
```

Comment:
```suggestion
    while logSink.CallCount < 3 do Threading.Thread.Sleep 5
```

Reply:
Even better though:
```
    do! src.Monitor.AwaitCompletion(TimeSpan.FromMilliseconds 1, awaitFullyCaughtUp = true)
```
should wait until the tail is observed.
If you update the tail sleep interval to something really long on L125, this `AwaitCompletion` should stop immediately (and the Stop on L131 can abort the sleep)

I'd also consider reducing the page slze (1000) on L125 in order to force multiple pages (which would have caused the total items to be off by one per such batch)